Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/HoughTransformSeeder.hpp

```diff
@@ -0,0 +1,279 @@
+// Each hit will have its own line based on its position (phi and r).
+// However, note that hits belonging to the same track will have lines that
+// intersect at the track's q/pT and phi. In this manner, we can conduct pattern
+// -matching by looking for intersections of these pT-phi lines.
+//
+// In other words, given some assumed q/pT for the track one can take the phi and r for a hit and convert that to 
+// what the phi(perigee) for a track must have been. We loop over the q/pT bins, and at the true value all the hits should line
+// up in the same bin
+//
+// To easily find intersections, we first pixelate (equivalently, we make a 2d
+// histogram from) the graph of all the hit's lines in q/pT vs phi_track space.
+// We then apply a convolution (i.e. a scanning window) to pick out points with
+// multiple lines going through them. These points become our seed.
+//
+// In principle the Hough transform can be used for an entire region (i.e. .2
+// phi x .2 eta) or larger. However this can lead to an excessive number of
+// hits/lines in the transform houghHist, leading to spurious intersections.
+// Instead, we can use multiple transforms that each cover a slice in z0, and
+// simply combine all the seeds found. These are the subregions
+//
+// References:
+//      Martensson Thesis:
+//      http://uu.diva-portal.org/smash/get/diva2:1341509/FULLTEXT01.pdf
+//
+
+// We adopt the following nomenclature within this class:
+//      houghHist: The 'graph' in q/pT vs phi_track space, filled with a line
+//      calculated as above for each hit. point: A specific q/pT and phi_track
+//      bin in the above houghHist; i.e. what is normally called a pixel
+//             but I don't want to confuse this with the detector type. A
+//             point's value is the number of lines that go through it.
+//
+// For the first iteration, x refers to phi_track, and y refers to q/pT,
+// although this should remain flexible. These are set via the variables m_par_x
+// and m_par_y.
+//
+// NOTE: y=0 represents the lowest q/pT bin, x=0 represents the lowest phi(perigee) bin
+//      houghHist[y=0][x=0]      : lowest q/pT and lowest phi_track bin
+//      houghHist[y=size-1][x=0] : highest q/pT and lowest phi_track bin
+//
+
+#pragma once
+
+#include "Acts/Geometry/GeometryIdentifier.hpp"
+#include "Acts/Utilities/Result.hpp"
+#include "Acts/Utilities/Delegate.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include "ActsExamples/EventData/Measurement.hpp"
+#include "ActsExamples/EventData/SimSpacePoint.hpp"
+#include "ActsExamples/Framework/BareAlgorithm.hpp"
+#include "ActsExamples/TrackFinding/HoughVectors.hpp"
+
+#include <string>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+using ResultDouble = Acts::Result<double>;
+using ResultBool = Acts::Result<bool>;
+using ResultUnsigned = Acts::Result<unsigned>;
+
+using FieldCorrector = Acts::Delegate<ResultDouble(unsigned, double, double)>; // (unsigned region, double y, double r)
```

Comment:
Nice, this is really great.

Reply:
Can you expand a bit more what you mean about the B field and using a Projector type struct, ie point me to something specific? I think I'm not sure what you're implying. I tried looking at the code and I see (in MultiTrajectory.hpp):
```
 using Projector =
      Eigen::Matrix<typename Covariance::Scalar, M, eBoundSize, ProjectorFlags>;
```
but I don't think that's what you mean?